cargo-paradedb/src/bench_hits.rs

```diff
@@ -0,0 +1,345 @@
+#![allow(unused_imports)]
+use anyhow::{anyhow, bail, Result};
+use cmd_lib::*;
+use sqlx::postgres::PgConnectOptions;
+use sqlx::{Connection, Executor, PgConnection};
+use std::fs::File;
+use std::io::BufRead;
+use std::path::{Path, PathBuf};
+use std::str::FromStr;
+use std::time::Instant;
+use std::{env, io};
+use tempfile::TempDir;
+
+fn download_and_verify(url: &str, checksum: &str, filename: &str) -> Result<()> {
+    if Path::new(filename).exists() {
+        run_cmd!(echo "Dataset '$filename' already exists, verifying checksum...")?;
+        if run_cmd!(echo "$checksum $filename" | md5sum -c --status).is_ok() {
+            run_cmd!(echo "Dataset '$filename' already exists and is verified, skipping download...")?;
+            return Ok(());
+        } else {
+            run_cmd!(echo "Checksum mismatch. Re-downloading '$filename'...")?;
+        }
+    }
+
+    run_cmd!(echo "Downloading $filename dataset...")?;
+    run_cmd!(wget --no-verbose --no-check-certificate --continue -O $filename $url)?;
+    Ok(())
+}
+
+pub async fn bench_hits(url: &str, tag: &str, workload: &str) -> Result<()> {
+    let _os = run_fun!(uname)?;
+
+    println!("\n*********************************************************************************");
+    println!(
+        "* Benchmarking pg_lakehouse version '{}' against ClickBench",
+        tag
+    );
+    println!("*********************************************************************************\n");
+
+    let root_file_path = PathBuf::from_str("/tmp")?;
+    let single_file_path = root_file_path.join("hits.parquet").display().to_string();
+    let partitioned_dir_path = root_file_path.join("partitioned/").display().to_string();
+
+    if workload == "single" {
+        run_cmd!(echo "Using ClickBench's single Parquet file for benchmarking...")?;
+        download_and_verify(
+            "https://paradedb-benchmarks.s3.amazonaws.com/hits_5m_rows.parquet",
```

Comment:
Currently, you just download the smallest dataset. We should have a flag, just like we do `-w partitioned/single` to decide whether to download the full or a small dataset, this way I can run the benchmarks in CI with the small dataset, and run the full dataset manually on an EC2 instance/dev laptops as needed

Reply:
Is `partitioned` always the full dataset?

I had added a `full` flag... but realized that was awkward because it has no effect with `partitioned`. Maybe we should have three options for `-w`... `small`, `full`, `partitioned`?

Reply:
That's not the case. Partitioned also has a 5m row dataset

`/partitioned/` is the full dataset, `/partitioned_5m_rows/` is the small dataset.

Reply:
Got it, it makes sense to add back the `full` flag then. I just pushed it, could you double check that I have all the URLS right?

Reply:
This is perfect, thank you!